src/features/pagination/test/pagination.spec.js

```diff
@@ -64,6 +64,8 @@ describe('ui.grid.pagination uiGridPaginationService', function () {
   describe('initialisation', function () {
     it('registers the API and methods', function () {
       expect(gridApi.pagination.getPage).toEqual(jasmine.any(Function));
+      expect(gridApi.pagination.getFirstRowIndex).toEqual(jasmine.any(Function));
```

Comment:
I've added tests for custom pagination.

Rather than testing `getFirstRowIndex` and `getLastRowIndex` directly, we test the resulting pages.

The previous tests covered one branch inside the two methods, and the tests I've added cover the other branch.